scene/3d/physics/collision_shape_3d.cpp

```diff
@@ -87,11 +94,18 @@ void CollisionShape3D::_notification(int p_what) {
 				if (shape.is_valid()) {
 					collision_object->shape_owner_add_shape(owner_id, shape);
 				}
+
 				_update_in_shape_owner();
 			}
 		} break;
 
 		case NOTIFICATION_ENTER_TREE: {
+#ifdef DEBUG_ENABLED
+			if (debug_color == Color(0.0, 0.0, 0.0, 0.0)) {
+				set_debug_color(get_default_debug_color());
```

Comment:
Somewhat nasty, given this snippet is repeated in `set_debug_color()` itself. You may want to have a separate function that does this specific check and updates the debug color.

Reply:
I've reworked a lot of situations like this to reduce the many questionable copy-pastes throughout the class. The code for ensuring the "placeholder" colour value resets to the default collision colour as needed is much simpler now.